src/blockdiagonal.jl

```diff
@@ -9,7 +9,6 @@ struct BlockDiagonal{T, V<:AbstractMatrix{T}} <: AbstractMatrix{T}
     blocks::Vector{V}
 
     function BlockDiagonal{T, V}(blocks::Vector{V}) where {T, V<:AbstractMatrix{T}}
-        all(is_square, blocks) || throw(ArgumentError("All blocks must be square."))
```

Comment:
The docstring will need updating

And I think the `is_square` function can be entirely removed?

Reply:
I think having the function may be useful for the methods that are exclusive to square matrices, such as inverse, power, etc (if wanted to add these methods later).